### PR TITLE
26 mariabackup backup method

### PR DESCRIPTION
**Pull request by kobybr**

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -36,14 +36,15 @@
...
-    $home    = '/var/lib/mysql'
-    $shell   = '/sbin/nologin'
-    $group   = 'mysql'
-    $groups  = undef
+    $user      = 'mysql'
+    $comment   = 'MySQL server'
+    $uid       = 494
+    $gid       = 494
+    $home      = '/var/lib/mysql'
+    $backupdir = '/var/lib/mysqlbackups'
```
Comment (edestecd) [inline manifests/params.pp]: why did we choose this as the default backup dir?

File: manifests/backup/mariabackup.pp [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+class mariadb::backup::mariabackup (
+  $mariabackup_package_name = $mariadb::params::backup_package_name,
+  $backupuser              = undef,
+  $backuppassword          = undef,
+  $backupdir               = $mariadb::params::backupdir,
+  $backupmethod            = 'mariabackup',
+  $backupdirmode           = '0700',
+  $backupdirowner          = $mariadb::params::user,
+  $backupdirgroup          = $mariadb::params::group,
```
Comment (edestecd) [inline manifests/backup/mariabackup.pp]: I'm not sure about using the mysql user/group. The cron runs as root and the xtrabackup uses root: https://github.com/puppetlabs/puppetlabs-mysql/blob/master/manifests/backup/xtrabackup.pp#L14

File: manifests/backup/mariabackup.pp [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  $backupcompress          = true,
+  $initiator_node          = false,
```
Comment (edestecd) [inline manifests/backup/mariabackup.pp]: do we need initiator node logic?

File: manifests/backup/mariabackup.pp [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    owner  => $backupdirowner,
+    group  => $backupdirgroup,
+  }
+
+  file { 'mariabackup.sh':
+    ensure  => $ensure,
+    path    => '/usr/local/sbin/mariabackup.sh',
+    mode    => '0700',
+    owner   => $backupdirowner,
+    group   => $backupdirgroup,
```
Comment (edestecd) [inline manifests/backup/mariabackup.pp]: why not root owner and group? the cron runs as root.

Comment (edestecd) [reply manifests/backup/mariabackup.pp]: https://github.com/puppetlabs/puppetlabs-mysql/blob/master/manifests/backup/xtrabackup.pp#L83

File: manifests/params.pp (hunk 1 above)
Comment (kobybr) [reply manifests/params.pp]: I found that if you put the backup dir in /var/lib/mysql.  It shows up as a database in show databases;    Doesn't seem like a good place to put it for that reason.

File: manifests/backup/mariabackup.pp (hunk 3 above)
Comment (kobybr) [reply manifests/backup/mariabackup.pp]: Without it we have to install cron on every node

File: manifests/backup/mariabackup.pp (hunk 2 above)
Comment (kobybr) [reply manifests/backup/mariabackup.pp]: Changed to root

File: templates/backup/mariabackup.sh.erb [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  <%- [@prescript].flatten.compact.each do |script| %>
+<%= script %>
+  <%- end -%>
+<% end -%>
+
+now=$(date)
+dayofweek=$(date +%w --date="${now}")
+weekofyear=$(date +%Y-%U --date="${now}")
+targetdir="<%= @backupdir -%>/${weekofyear}/${dayofweek}"
+<%- unless @never_incremental == true -%>basedir="<%= @backupdir -%>/${weekofyear}/$[dayofweek - 1]"<% end %>
```
Comment (edestecd) [inline templates/backup/mariabackup.sh.erb]: could probably clean up these unless @never_incremental

**Pull request merged**